WordPress/src/main/java/org/wordpress/android/ui/notifications/adapters/NotesAdapter.java

```diff
@@ -168,6 +168,11 @@ private Note getNoteAtPosition(int position) {
         return null;
     }
 
+    public void replaceNote(Note newNote) {
+        int position = getPositionForNoteUnfiltered(newNote.getId());
+        mNotes.set(position, newNote);
```

Comment:
We should check that `position  != recyclerView.NO_POSITION` and just to be super sure `position <  mNotes.size()`